### ISSUE TITLE
ern run-android fails in Mac OS

### ISSUE THREAD

**Issue by faisalmohd83** [electrode-io/MovieListMiniApp#26]
As per the [getting-started](https://native.electrode.io/v/v0.35/quick-start/getting-started) sample app, the `ern run-android`  command should create an android project within the MiniApp and run. Rather the same is resulting as below:

Is it due to any missing steps or dependencies?

**ERROR**:

>  An error occurred: Command failed: ./gradlew lib:uploadArchives
> 
>  FAILURE: Build failed with an exception.
> 
>  * What went wrong:
>  Could not initialize class org.codehaus.groovy.runtime.InvokerHelper
> 
>  * Try:
>  Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full in
sights.
>  
>  * Get more help at https://help.gradle.org
>  
>  BUILD FAILED in 0s

Environment(s):

- OS: macOS Big Sur
- Java: OpenJDK version "14.0.1" 2020-04-14
- Gradle: Gradle 6.8.3

--

- electrode: v0.45.6
- npm: 6.14.11
- node: v14.16.0
- yarn: 1.22.10

**Comment by faisalmohd83** [comment]
Solved!

Couple of issues noticed, would require a manual fix to get it work for Android.

1. Java version downgraded from 14 to 11. (Make sure gradle is >6.5)
2. NDK local path to be mentioned in local.properties

The above worked for me to solve in my local workspace.

**Comment by friederbluemle** [comment]
Thanks for the update @faisalmohd83 :+1:

Regarding the Java version: For Android projects, it's generally best to use the jdk that comes with Android Studio. For the current version that would be `openjdk version "1.8.0_242-release"`.

On macOS:

```
/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home/bin/java -version
```

**Comment by faisalmohd83** [comment]
@friederbluemle

Is there any reason electrode sticks to Java11?
My sample electrode app was referring to system-level (on macOS) Java/Gradle rather than AndroidStudio one.

**Comment by friederbluemle** [comment]
The Java version for Android has nothing at all to do with Electrode Native. The distribution that is bundled with Android Studio should be considered the "official" version to use for Android projects.
I'm surprised you're able to build your project with Java 11.. Is there a specific reason?

**Comment by faisalmohd83** [comment]
So, My Q is why does the EN project rely on AndroidStudio rather than fetching from device default versions? let it Java, gradle or react!

--
OK, So what I did was install Java11 and set the macOS JAVA_HOME path in ~/.zshrc to Java11. With this, I could compile the `ern run-android` and finally the `android/` folder showed up in the sample App. Then the Android project opened in AndroidStudio and tried to run which resulted in no ndk mentioned! ( I really wonder is there any requirement for NDK in the whole project). Provinding ndk.dir in local.properties worked and code compiled.

long story short, it wasn't straight forward for Android whereas iOS project was simple.
